matchms/filtering/metadata_processing/repair_adduct_based_on_parent_mass.py

```diff
@@ -0,0 +1,70 @@
+import logging
+from matchms import Spectrum
+from ..filter_utils.load_known_adducts import load_known_adducts
+
+
+logger = logging.getLogger("matchms")
+
+
+def repair_adduct_based_on_parent_mass(spectrum_in: Spectrum,
+                                       mass_tolerance: float):
+    """
+    Corrects the adduct of a spectrum based on its parent_mass representation and the precursor m/z.
+
+    Parameters:
+    ----------
+    spectrum_in : Spectrum
+        The input spectrum whose adduct needs to be repaired.
+
+    mass_tolerance : float
+        Maximum allowed mass difference between the parent mass and the parent mass based on the adduct.
+    """
+    if spectrum_in is None:
+        return None
+    changed_spectrum = spectrum_in.clone()
+
+    new_adduct = _get_matching_adduct(precursor_mz=spectrum_in.get("precursor_mz"),
+                                      parent_mass=spectrum_in.get("parent_mass"),
+                                      ion_mode=spectrum_in.get("ionmode"),
+                                      mass_tolerance=mass_tolerance)
+    if new_adduct is None:
+        return spectrum_in
+
+    changed_spectrum.set("adduct", new_adduct)
+    logger.info("Adduct was set from %s to %s",
+                spectrum_in.get('adduct'), new_adduct)
+    return changed_spectrum
+
+
+def _get_matching_adduct(precursor_mz, parent_mass, ion_mode, mass_tolerance):
+    if precursor_mz is None:
+        logger.warning("Precursor_mz is None, first run add_precursor_mz")
+        return None
+
+    if ion_mode not in ("positive", "negative"):
+        if ion_mode is not None:
+            logger.warning("Ionmode: %s not positive, negative or None, first run derive_ionmode",
+                           ion_mode)
```

Comment:
```suggestion
    if ion_mode not in ("positive", "negative", None):
        logger.warning("Ionmode: %s not positive, negative or None, first run derive_ionmode",
                           ion_mode)
```

Reply:
@florian-huber I think this would not work. Because I want to return None if ionmode is None. Let me know if you think we should still change it. For now I will merge the PR. But I can quickly make a new one to add the change again later.